### PR TITLE
feat(hybrid-cloud): Add a test for file upload

### PR DESCRIPTION
**Pull request by AniketDas-Tekky**
This just ensures that the file body is in the request body that's being proxied

I'm kinda at an impasse here because I need to see what's coming into the django view from the proxied request. Parsing the raw request object is strange because I can't seem to extract the input buffer

### CODE REVIEW

File: src/sentry/api_gateway/proxy.py [hunk 1]
```diff
@@ -45,12 +45,15 @@ def proxy_request(request: Request, org_slug: str) -> StreamingHttpResponse:
     request_args = {
         "headers": request.headers,
         "params": dict(query_params) if query_params is not None else None,
-        "data": getattr(request, "body", None),
+        "files": getattr(request, "FILES", None),
+        "data": getattr(request, "body", None)
+        if request.content_type == "application/json"
```
Comment (markstory) [inline src/sentry/api_gateway/proxy.py:50]: We might also have endpoints (especially in django views) that are taking form encoded POST bodies. Should we pass those through as well?

Comment (AniketDas-Tekky) [reply src/sentry/api_gateway/proxy.py:50]: If we don't then those pages will break right? I guess right now I'm not really seeing a reason why not to?

Comment (markstory) [reply src/sentry/api_gateway/proxy.py:50]: Wouldn't the current logic not forward data for form-encoded requests? Request data is only forwarded for json requests.

Comment (AniketDas-Tekky) [reply src/sentry/api_gateway/proxy.py]: Updated.

**Review by RyanSkonnord — APPROVED** [review]
LGTM

**Pull request merged**